### PR TITLE
Hydra MTurk fixes, plus doc update

### PR DESCRIPTION
**Pull request by JackUrb**
# Overview
A small bug was flagged in #256 that was breaking deploys to MTurk sandbox (as well as an unexpected change following applying Omry's advice in #250). This updates the Doc describing how to work things in the quickstart, and fixes the underlying bug in the mturk provider.

# Testing
Screen Shot 2020-09-21 at 9 30 59 AM
`pytest` runs tests fine.

### CODE REVIEW

File: docs/quickstart.md [hunk 1]
```diff
@@ -83,8 +83,9 @@ $ python static_test_script.py mephisto.architect=heroku mephisto.provider.reque
 # Note: my_mturk_user_sandbox is what we used to name the requester
 # when we registered the mturk account in the previous step.
 ```
-The arguments `mephisto.provider.requester_name=my_mturk_user_sandbox` and `mephisto.architect=heroku` will tell the the script to use the mturk sandbox provider and the heroku architect (as opposed to the mock provider and local architect).
+The arguments `mephisto.provider.requester_name=my_mturk_user_sandbox` and `mephisto/architect=heroku` will tell the the script to use the mturk sandbox provider and the heroku architect (as opposed to the mock provider and local architect). Notice that if we're setting a full abstraction (like the architect) we reference it with `mephisto/abstraction=val`, however when we're setting an argument, we use `mephisto.abstraction.argument=val`. This tells hydra whether we're providing an argument value or the name of a configuration to load.
```
Comment (EricMichaelSmith) [inline docs/quickstart.md:86]: Hmm - line 81 should now be changed to say `mephisto/architect=heroku`, right?

**Review by EricMichaelSmith — APPROVED** [review]
Seems sensible! Stamping, but others might have other concerns

File: docs/quickstart.md (hunk 1 above)
Comment (EricMichaelSmith) [reply docs/quickstart.md:86]: Also, there's still a reference to `mephisto.architect=heroku` in https://github.com/facebookresearch/Mephisto/blob/master/docs/hydra_migration.md

Comment (JackUrb) [reply docs/quickstart.md:86]: Thanks for the flags! Updated all of them now

**Pull request merged**